### PR TITLE
Add Progress thumb size

### PR DESCRIPTION
**Pull request by GDmail91**
I want to set thumb size of each circular progress bars.
So I splitted the thumb size and progress thickness.

I've added default thumb size (as 10dp) and thumb & shadow Paint object.

---
I cannot find other branch. So I pushed in master branch.

### CODE REVIEW

**Comment by GuilhE** [comment]
Hello @GDmail91 and thanks for your contribution.
It's a nice feature but you forgot to handle 2 corner cases:

1. `onMeasure`:

Screenshot 2020-07-26 at 14 07 10

2. Compatibility

A flag `autoThumb` (`true` by default): the thumb changes it's size according to progress thickness.
If `autoThumb` is `false`, user is responsible for adjusting the thumbSize independently progressThickness.
This way we won't break compatibility.

**Comment by GuilhE** [comment]
Hello again @GDmail91 , thanks for you changes.
We're almost done, just 2 minor changes:
1. Remove **Thumb.java** since it's not being used;
2. Please add` { } `in **if-else** conditions, example:
 ```
if (mProgressThumbScaleType == POINT)
    mProgressThumbSize = size;
else
    mProgressThumbSize = Math.max(Math.min(size, mMaxThumbSizeRate), 0); // To prevent the Thumb size too big
```
to:
```
 if (mProgressThumbScaleType == POINT) {
     mProgressThumbSize = size;
 } else {
     mProgressThumbSize = Math.max(Math.min(size, mMaxThumbSizeRate), 0); // To prevent the Thumb size too big
 }
```

Thanks.

**Comment by GDmail91** [comment]
I Fixed  above issues and also auto & rate scaleType calculate issue

You can see the whole feature that I've added in sample app

Here is some description

1. Changed progress thumb size 

ProgressThumbScaleType.AUTO : It will be same as before
ProgressThumbScaleType.POINT : It will change the Thumb size by dp 
ProgressThumbScaleType.RATE : It will change the Thumb size by rate of stroke

Default Progress Scale Type is AUTO

```xml
... 
<com.github.guilhe.views.CircularProgressView
                android:id="@+id/sample_CircularProgressView"
                android:layout_width="@dimen/progress_size"
                android:layout_height="@dimen/progress_size"
                app:layout_constraintEnd_toEndOf="parent"
                app:layout_constraintStart_toStartOf="parent"
                app:layout_constraintTop_toTopOf="parent"
                app:progress="100"
                app:progressThumbScaleType="point"
                app:progressThumbSize="35dp"
                app:progressBarColorArray="@array/rainbow"
                app:progressBarRounded="true"
                app:progressThumb="false" />


```

**Comment by GDmail91** [comment]
+ 
I think there is no rate size setting by xml 
I will added it soon

**Comment by GuilhE** [comment]
> I think there is no rate size setting by xml
> I will added it soon

Ok thanks!
I'll add some comments in code where you missed `{ }` 😉

File: sample/src/main/java/sample/SampleActivity.java [hunk 1]
```diff
@@ -95,6 +104,14 @@ public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
             case R.id.thickness_SeekBar:
                 mBinding.sampleCircularProgressView.setProgressStrokeThickness((int) (progress * getResources().getDisplayMetrics().density + 0.5f));
                 break;
+            case R.id.thumbsize_SeekBar:
+                float size;
+                if (mBinding.sampleCircularProgressView.getProgressThumbScaleType() == ProgressThumbScaleType.RATE)
```
Comment (GuilhE) [inline sample/src/main/java/sample/SampleActivity.java:109]: Please wrap in `{ }`

File: circular-progress-view/src/main/java/com/github/guilhe/views/CircularProgressView.java [hunk 2]
```diff
@@ -604,7 +676,19 @@ protected synchronized void onDraw(Canvas canvas) {
...
-        float radius = getWidth() / 2 - mDefaultViewPadding - mProgressIconThickness - mProgressStrokeThickness / 2;
+        float radius = (float) getWidth() / 2 - mDefaultViewPadding - mProgressIconThickness - mProgressStrokeThickness / 2;
+
+        float thumbSize = 0;
+        if (mProgressThumbScaleType == POINT) {
+            thumbSize = mProgressThumbSize;
+            radius -= (thumbSize - mProgressStrokeThickness / 2);
+        } else if (mProgressThumbScaleType == RATE) {
+            thumbSize = (mProgressStrokeThickness / 2) * mProgressThumbSize;
+            if (mProgressThumbSize > 1)
```
Comment (GuilhE) [inline circular-progress-view/src/main/java/com/github/guilhe/views/CircularProgressView.java:687]: @GDmail91 Please wrap in `{ }`

File: circular-progress-view/src/main/java/com/github/guilhe/views/CircularProgressView.java [hunk 3]
```diff
@@ -572,11 +623,32 @@ protected synchronized void onMeasure(int widthMeasureSpec, int heightMeasureSpe
...
+            thumbSize = mProgressStrokeThickness;
+        }
+
+        // if ThumbSize diameter is thicker than Stroke
+        if (mProgressThumbEnabled) {
+            // increase progress width
+            if (mProgressThumbScaleType == POINT && thumbSize * 2 > mProgressStrokeThickness) {
+                progressWidth += thumbSize - mProgressStrokeThickness + mProgressStrokeThickness / 2;
+            } else if (mProgressThumbScaleType == RATE) {
+                if (thumbSize > mProgressStrokeThickness)
```
Comment (GuilhE) [inline circular-progress-view/src/main/java/com/github/guilhe/views/CircularProgressView.java:643]: @GDmail91 Please wrap in `{ }`

**Comment by GuilhE** [comment]
Thanks for your contribution @GDmail91  👍

**Pull request merged**